radixdlt-core/radixdlt/src/main/java/com/radixdlt/statecomputer/forks/Forks.java

```diff
@@ -247,6 +247,24 @@ public void init(CommittedReader committedReader, ForksEpochStore forksEpochStor
     sanityCheckStoredForksAgainstConfiguration(forksEpochStore, currentEpoch);
   }
 
+  private long getCurrentEpoch(CommittedReader committedReader) {
+    final var lastRecordedProofOptional = committedReader.getLastProof();
```

Comment:
I believe `Opt` suffix is good enough.

Reply:
What do you think about?

```
final var currentEpoch =
        committedReader
            .getLastProof()
            .map(
                lastProof ->
                    lastProof.isEndOfEpoch() ? lastProof.getEpoch() + 1 : lastProof.getEpoch())
            .orElse(0L);
```

Reply:
or even better in a Kotlin style

```
final var currentEpoch =
        committedReader
            .getLastProof()
            .map(it -> it.isEndOfEpoch() ? it.getEpoch() + 1 : it.getEpoch())
            .orElse(0L);
```

Reply:
A part of me likes it, a part of me doesn't want to hide this logic in a map - because I want to add comments etc :)

Reply:
You can still add comments. I posted a version with comments, but removed it as the code is quite straightforward IMO.